school/config/routes.rb

```diff
@@ -1,3 +1,17 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get '/courses/new' => 'courses#new'
+  post '/courses' => 'courses#create'
+  get '/courses' => 'courses#index'
+  get '/courses/:id' => 'courses#edit'
+  post '/courses/edit' => 'courses#update'
+  get '/courses/delete' => 'courses#delete'
+  get '/courses/show' => 'courses#show'
```

Comment:
There is one small problem with your `show` and `delete` routes. When I try to view the course show page or try to delete a course I get a `NoMethodError` 
<img width="646" alt="screen shot 2019-02-02 at 3 15 38 pm" src="https://user-images.githubusercontent.com/8661078/52168871-8f0bb000-26fd-11e9-9e65-5eb826b48f86.png">

Your route in this case is confused and trying to point to your edit action. 
If you want to follow the Rails convention of doing routes you can read up [here](https://guides.rubyonrails.org/routing.html)

How about trying something like

`get /contacts/:id  => contacts#show` display a specific contact <-- `:id` will be the contact id, it's so your route knows which contact to render on the show page.
`get /contacts/:id/edit => contacts#edit` <-- return an HTML form for editing a specific contact (edit.html.erb)
`PATCH/PUT  /contacts/:id => contacts#update` <-- pass in id to update a specific contact
`DELETE | /contacts/:id | contacts#destroy` <-- Your delete action expects an id that way it know which contact to delete

That way when you hit your actions you can access the contact id using `Course.find_by(id: params[:id])` in your controller, which you are already doing!

Reply:
Sorry, in your routes i meant `courses` not `contacts` lol